test/to-json.test.js

```diff
@@ -1,11 +1,41 @@
 'use strict'
 
-const { test } = require('tap')
+const { test } = require('node:test')
 const boot = require('..')
 
-test('to json', (t) => {
-  t.plan(4)
+function matchObject (t, actual, expected) {
+  const isExpectedArray = Array.isArray(expected)
+  const isExpectedRegex = expected instanceof RegExp
 
+  if (typeof expected === 'object' && expected !== null && !isExpectedArray && !isExpectedRegex) {
+    for (const key in expected) {
+      if (!(key in actual)) {
+        throw new Error(`Missing key: ${key}`)
+      }
+      matchObject(t, actual[key], expected[key])
+    }
+    return
+  }
+
+  if (isExpectedArray) {
+    t.assert.ok(Array.isArray(actual))
+    t.assert.strictEqual(actual.length, expected.length)
+    for (let i = 0; i < expected.length; i++) {
+      matchObject(t, actual[i], expected[i])
+    }
+    return
+  }
+
+  if (isExpectedRegex) {
+    t.assert.match(actual.toString(), expected)
+    return
+  }
+
+  t.assert.strictEqual(actual, expected)
+}
```

Comment:
what is this?

Reply:
node test runner doesn't implement match in the same way tap does.
Too keep the `outJson` structure, I've created this companion function that mimics the behaviour of tap's match.